Unit testing best practices (#1381)
demisto/content-docs: docs/integrations/unit-testing.md

If the function raises a ValueError with proper error message, the test will pass.

## Content Testing Guidelines

* When testing a command, mock all external requests (e.g. mock `Client` methods), run the command on given inputs, and compare the output to a hardcoded value.
* Prefer comparing whole outputs (e.g. comparing the output to a `CommandResult` object), to make sure the code behavior is preserved even when others add features and fix issues.

> * Use the `@freezegun.freeze_time` decorator on the testing method when the output may rely on the time of running.

Reviewer: ```suggestion
* Use the `@freezegun.freeze_time` decorator on the testing method when the output may rely on the time or date (think Summer Time) of running.
```

Author: ```suggestion
* Use the `@freezegun.freeze_time` decorator on the testing method when the output may vary in different time, date or timezone (e.g. daylight-saving time) in run time.
```

* Methods that handle errors or exceptions should have multiple tests - one for the _happy path_ (normal functionality, without errors), and one testing method for each error handled (a.k.a _sad path/s_).

> 

Reviewer 2: ```suggestion
* Don't include any sensitive data (secrets, tokens, email addresses, personal information, etc.).
```

Author: ```suggestion
* Make sure to sanitize or remove any sensitive data (passwords, tokens, email addresses, personal information, etc.) from unit tests and test data.

```

If the function raises a ValueError with proper error message, the test will pass.

## Content Testing Guidelines

> 

Reviewer 3: would add adise to mock demistomock directly using mocker, not rely on what we have in the file

If the function raises a ValueError with proper error message, the test will pass.

## Content Testing Guidelines

> * When testing a command, mock all external requests (e.g. mock `Client` methods), run the command on given inputs, and compare the output to a hardcoded value.

Reviewer 3: ```suggestion
* When testing a command, mock all external requests (e.g. mock `Client` methods), run the command on given inputs, and compare the output to a hardcoded value.
* When using demistomock, mock demisto directly using *mocker*, don't rely on the defaults.
```

Author: ```suggestion
* When testing a command, mock all external requests (e.g. mock `Client` methods), run the command on given inputs, and compare the output to a hardcoded value.
* When using demistomock, mock the `demisto` object directly using `mocker`, don't rely on the defaults.
```
